tests/test_validate_parameters.py

```diff
@@ -0,0 +1,53 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of REANA.
+# Copyright (C) 2021 CERN.
+#
+# REANA is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""REANA client validate parameters tests."""
+
+import yaml
+
+from reana_client.validation import validate_serial_parameters
+
+
+def test_validate_parameters_serial(create_yaml_workflow_schema, capsys):
+    """Validate parameters for Serial workflows."""
+    reana_yaml = yaml.load(create_yaml_workflow_schema, Loader=yaml.FullLoader)
+    reana_yaml["inputs"]["parameters"]["foo"] = "foo"
+    validate_serial_parameters(reana_yaml)
+    captured = capsys.readouterr()
+    assert 'WARNING: Input parameter "foo" is not being used' in captured.out
+    del reana_yaml["inputs"]["parameters"]["foo"]
+
+    reana_yaml["workflow"]["specification"]["steps"].append(
+        {"commands": ["python ${foo} --bar"]}
+    )
+    validate_serial_parameters(reana_yaml)
+    captured = capsys.readouterr()
+    assert (
+        'WARNING: Parameter "foo" found on step "Unnamed" is not defined in inputs parameters'
```

Comment:
Perhaps, instead of "Unnamed" we can simply display the step number that has the error? It is not fully harmonised with the UI (there if the step doesn't have a name, we display the job UUID, see below) but still it could make more sense?
![image](https://user-images.githubusercontent.com/9035606/109812700-abf07580-7c2c-11eb-83b3-975e77eaf545.png)

Reply:
How can we get this on the client-side?

Reply:
Nevermind, I think I hadn't understood your message... Updated.